internal/engine/configs/configs_controller_test.go

```diff
@@ -73,11 +73,49 @@ func TestConfigsControllerDockerNotConnectedButNotRequired(t *testing.T) {
 	bar := manifestbuilder.New(f, "bar").
 		WithK8sYAML(testyaml.SanchoYAML).
 		Build()
-	a := f.run(bar)
+	_, a := f.run(bar)
 
 	assert.NoError(t, a.Err)
 }
 
+func TestBuildReasonTrigger(t *testing.T) {
+	f := newCCFixture(t)
+	defer f.TearDown()
+
+	f.addManifest("fe")
+	bar := manifestbuilder.New(f, "bar").WithK8sYAML(testyaml.SanchoYAML).Build()
+	f.setTiltfileHasBuilt()
+
+	state := f.st.LockMutableStateForTesting()
+	state.AppendToTriggerQueue(model.TiltfileManifestName, model.BuildReasonFlagTriggerWeb)
+	f.st.UnlockMutableState()
+
+	reloadStarted, _ := f.run(bar)
+
+	assert.Equal(t, model.BuildReasonFlagTriggerWeb, reloadStarted.Reason)
+}
+
+func TestBuildReasonTriggerAndOtherReason(t *testing.T) {
+	f := newCCFixture(t)
+	defer f.TearDown()
+
+	f.addManifest("fe")
+	bar := manifestbuilder.New(f, "bar").WithK8sYAML(testyaml.SanchoYAML).Build()
+	f.setTiltfileHasBuilt()
+
+	state := f.st.LockMutableStateForTesting()
+	state.AppendToTriggerQueue(model.TiltfileManifestName, model.BuildReasonFlagTriggerWeb)
+	state.PendingConfigFileChanges["somefile.txt"] = time.Now()
```

Comment:
If you want this test to work as written, you probably don't want to use time.Now() here, due to the monotonic clocks issue i mentioned earlier. simplest solution is to use `time.Now().Add(- time.Second)` here instead